plugins/modules/ec2_instance.py

```diff
@@ -1630,6 +1630,41 @@ def value_wrapper(v):
     return changes_to_apply
 
 
+def change_instance_metadata_options(instance, params):
+    changed = False
+
+    existing_metadata_options = camel_dict_to_snake_dict(instance.get("MetadataOptions"))
+    metadata_options_to_apply = params.get("metadata_options")
```

Comment:
Rather than wrapping the initial call (line 1999) in an if statement we can simply return quickly if there's nothing to do.  This keeps all of the metadata_options logic together.

```suggestion
    metadata_options_to_apply = params.get("metadata_options")
    if metadata_options_to_apply is None:
        return False

    existing_metadata_options = camel_dict_to_snake_dict(instance.get("MetadataOptions"))
    ```